src/widgets/helper/ChannelView.cpp

```diff
@@ -273,12 +273,30 @@ std::pair<int, int> getWordBounds(MessageLayout *layout,
     return {wordStart, wordStart + length};
 }
 
+auto *requireNotNull(auto *ptr)
+{
+    assert(ptr != nullptr && "Pointer must not be nullptr");
+    return ptr;
+}
+
 }  // namespace
 
 namespace chatterino {
 
-ChannelView::ChannelView(BaseWidget *parent, Split *split, Context context,
+ChannelView::ChannelView(QWidget *parent, Context context, size_t messagesLimit)
+    : ChannelView(InternalCtor{}, parent, nullptr, context, messagesLimit)
+{
+}
+
+ChannelView::ChannelView(QWidget *parent, Split *split, Context context,
                          size_t messagesLimit)
+    : ChannelView(InternalCtor{}, requireNotNull(parent), requireNotNull(split),
```

Comment:
I would prefer if the asserts for these params are made in the constructor itself, without the `requireNotNull` function

Technically they're asserting too late, but since both values are allowed to be null by the internal ctor, that should be fine, and the assert will fire with a little bit better info